Source/SharpNeedle/HedgehogEngine/Needle/TextureStreaming/DataBlock.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.IO;
+
+namespace SharpNeedle.HedgehogEngine.Needle.TextureStreaming;
+
+public class DataBlock : IBinarySerializable
+{
+    public long Position { get; set; }
+
+    public long Length { get; set; }
+
+    public MemoryStream DataStream { get; private set; }
```

Comment:
The base `Stream` class should be preferred here instead of a MemoryStream. It'd allow for a SubStream to be assigned on read and when EnsureData is called the field would be set to a MemoryStream. This'd allow for streaming data easily instead of jumping around hoops.

Reply:
Adjusted